IdentityCore/src/cache/key/MSIDMacKeychainAccountCacheKey.m

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDMacKeychainAccountCacheKey.h"
+
+static NSString *keyDelimiter = @"-";
+static NSInteger kAccountTypePrefix = 1000;
+
+@implementation MSIDMacKeychainAccountCacheKey
+
+- (NSNumber *)type
+{
+    if (self.accountType)
```

Comment:
isn't this done in the parent class already?

Reply:
The `if()` isn't in the parent. Since for our usage we don't know the type beforehand, this was resulting in setting the key attribute to `1000`, which wouldn't match anything. In the latest PR updates I've eliminated the subclass and added the `if()` check to the base class.  Similarly, I added an `if()` for the account method. A `readAllAccounts:` would cause an account attribute string of `@"-"` and fail to match any accounts.

Reply:
Update: it's handled in the query. double check

Reply:
key/query usage updated